### PR TITLE
Adding configuration for the "secondFork" network upgrade (definitive name TBD)

### PR DESCRIPTION
**Pull request by pmprete**
- Added networks' individual configuration files
- Added block-based configuration (commented and TBD for testnet and mainnet)
- Added parameter-based configuration for secondFork in devnet and regtest

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 5 issues

1. ![MAJOR][MAJOR] [DevNetSecondForkConfig.java#L22](https://github.com/rsksmart/rskj/blob/a934c8e03fc21d0c7676b41acd5f52bdc7ba4c23/rskj-core/src/main/java/org/ethereum/config/blockchain/devnet/DevNetSecondForkConfig.java#L22): This class has 6 parents which is greater than 5 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [MainNetSecondForkConfig.java#L22](https://github.com/rsksmart/rskj/blob/a934c8e03fc21d0c7676b41acd5f52bdc7ba4c23/rskj-core/src/main/java/org/ethereum/config/blockchain/mainnet/MainNetSecondForkConfig.java#L22): This class has 6 parents which is greater than 5 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [TestNetSecondForkConfig.java#L22](https://github.com/rsksmart/rskj/blob/a934c8e03fc21d0c7676b41acd5f52bdc7ba4c23/rskj-core/src/main/java/org/ethereum/config/blockchain/testnet/TestNetSecondForkConfig.java#L22): This class has 6 parents which is greater than 5 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![INFO][INFO] [MainNetConfig.java#L40](https://github.com/rsksmart/rskj/blob/a934c8e03fc21d0c7676b41acd5f52bdc7ba4c23/rskj-core/src/main/java/org/ethereum/config/net/MainNetConfig.java#L40): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [TestNetConfig.java#L37](https://github.com/rsksmart/rskj/blob/a934c8e03fc21d0c7676b41acd5f52bdc7ba4c23/rskj-core/src/main/java/org/ethereum/config/net/TestNetConfig.java#L37): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rskj-core/src/main/java/org/ethereum/config/net/DevNetConfig.java [hunk 1]
```diff
@@ -40,6 +43,7 @@
     public static DevNetConfig getDefaultDevNetConfig() {
         DevNetConfig config = new DevNetConfig();
         config.add(0, new DevNetOrchidConfig());
+        config.add(0, new DevNetSecondForkConfig());
```
Comment (lsebrie) [inline rskj-core/src/main/java/org/ethereum/config/net/DevNetConfig.java]: This line it's ok, but we should erase the previous configuration for devnet

Comment (pmprete) [reply rskj-core/src/main/java/org/ethereum/config/net/DevNetConfig.java]: good catch, fixing it

**Comment by rskops** [comment]
SonarQube analysis reported 5 issues

1. ![MAJOR][MAJOR] [DevNetSecondForkConfig.java#L22](https://github.com/rsksmart/rskj/blob/4c122d5370d2f3ab96fc854c3e42dd92aef4aa85/rskj-core/src/main/java/org/ethereum/config/blockchain/devnet/DevNetSecondForkConfig.java#L22): This class has 6 parents which is greater than 5 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [MainNetSecondForkConfig.java#L22](https://github.com/rsksmart/rskj/blob/4c122d5370d2f3ab96fc854c3e42dd92aef4aa85/rskj-core/src/main/java/org/ethereum/config/blockchain/mainnet/MainNetSecondForkConfig.java#L22): This class has 6 parents which is greater than 5 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [TestNetSecondForkConfig.java#L22](https://github.com/rsksmart/rskj/blob/4c122d5370d2f3ab96fc854c3e42dd92aef4aa85/rskj-core/src/main/java/org/ethereum/config/blockchain/testnet/TestNetSecondForkConfig.java#L22): This class has 6 parents which is greater than 5 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![INFO][INFO] [MainNetConfig.java#L40](https://github.com/rsksmart/rskj/blob/4c122d5370d2f3ab96fc854c3e42dd92aef4aa85/rskj-core/src/main/java/org/ethereum/config/net/MainNetConfig.java#L40): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [TestNetConfig.java#L37](https://github.com/rsksmart/rskj/blob/4c122d5370d2f3ab96fc854c3e42dd92aef4aa85/rskj-core/src/main/java/org/ethereum/config/net/TestNetConfig.java#L37): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by rskops** [comment]
SonarQube analysis reported 5 issues

1. ![MAJOR][MAJOR] [DevNetSecondForkConfig.java#L22](https://github.com/rsksmart/rskj/blob/4d38175aa636a7aac8879bf26685ebc4dd3091cb/rskj-core/src/main/java/org/ethereum/config/blockchain/devnet/DevNetSecondForkConfig.java#L22): This class has 6 parents which is greater than 5 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [MainNetSecondForkConfig.java#L22](https://github.com/rsksmart/rskj/blob/4d38175aa636a7aac8879bf26685ebc4dd3091cb/rskj-core/src/main/java/org/ethereum/config/blockchain/mainnet/MainNetSecondForkConfig.java#L22): This class has 6 parents which is greater than 5 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [TestNetSecondForkConfig.java#L22](https://github.com/rsksmart/rskj/blob/4d38175aa636a7aac8879bf26685ebc4dd3091cb/rskj-core/src/main/java/org/ethereum/config/blockchain/testnet/TestNetSecondForkConfig.java#L22): This class has 6 parents which is greater than 5 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![INFO][INFO] [MainNetConfig.java#L40](https://github.com/rsksmart/rskj/blob/4d38175aa636a7aac8879bf26685ebc4dd3091cb/rskj-core/src/main/java/org/ethereum/config/net/MainNetConfig.java#L40): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [TestNetConfig.java#L37](https://github.com/rsksmart/rskj/blob/4d38175aa636a7aac8879bf26685ebc4dd3091cb/rskj-core/src/main/java/org/ethereum/config/net/TestNetConfig.java#L37): Complete the task associated to this TODO comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**